apps/api-gateway/src/issuance/issuance.controller.ts

```diff
@@ -507,6 +507,39 @@ async downloadBulkIssuanceCSVTemplate(
     };
     return res.status(HttpStatus.OK).json(finalResponse);
   }
+  @Get('/orgs/:orgId/:fileId/bulk/file-details-and-file-data')
+  @Roles(OrgRoles.ADMIN, OrgRoles.VERIFIER, OrgRoles.ISSUER, OrgRoles.OWNER)
+  @UseGuards(AuthGuard('jwt'), OrgRolesGuard)
+  @ApiBearerAuth()
+  @ApiUnauthorizedResponse({
+    status: HttpStatus.UNAUTHORIZED,
+    description: 'Unauthorized',
+    type: UnauthorizedErrorDto
+  })
+  @ApiResponse({ status: HttpStatus.OK, description: 'Success', type: ApiResponseDto })
+  @ApiForbiddenResponse({
+    status: HttpStatus.FORBIDDEN,
+    description: 'Forbidden',
+    type: ForbiddenErrorDto
+  })
+  @ApiOperation({
+    summary: 'Get all file details and file data by file id',
+    description: 'Get all file details and file data by file id'
+  })
+  async getFileDetailsAndFileDataByFileId(
+    @Param('orgId') orgId: string,
+    @Param(new ValidationPipe({ transform: true })) query: FileQuery,
+    @Res() res: Response
+  ): Promise<object> {
+    const { fileId } = query;
+    const issuedFileDetails = await this.issueCredentialService.getFileDetailsAndFileDataByFileId(orgId, fileId);
+    const finalResponse: IResponseType = {
+      statusCode: HttpStatus.OK,
+      message: ResponseMessages.issuance.success.fileDetailsAndFileData,
+      data: issuedFileDetails.response
```

Comment:
Using `sendNatsMessage` which I have mentioned in service function, you can simply use data: issuedFileDetails. As part of the new API refactorization task, we no longer include the response key in the data. Please refer to other functions for reference